docs/examples/viz_tesseract.py

```diff
@@ -0,0 +1,186 @@
+# e_color: color of the line actor (default: (1, 1, 1, 1))
+# dtheta: change in `angle` on each iteration. It determines the "speed" of the
+#        animation. Increase dtheta to increase speed of rotation, It may
+#        result in less smoother rotation (default: 0.02)
+# angle: defines the angle to be rotated to perform the animation, It changes
+#        as we run the `callback` method later on. (initial value: 0)
+
+p_color = np.array([0, 0.5, 1, 1])
+e_color = np.array([1, 1, 1, 1])
+dtheta = 0.02
+angle = 0
+
+###############################################################################
+# Let's define vertices for our 4D cube, `verts4D` contains the coordinates of
+# our 4D tesseract.
+
+verts3D = np.array(
+    [[1, 1, 1],
+     [1, -1, 1],
+     [-1, -1, 1],
+     [-1, 1, 1],
+     [-1, 1, -1],
+     [1, 1, -1],
+     [1, -1, -1],
+     [-1, -1, -1]]
+)
+
+# We can use primitive.box alternatively to get the cube's 3-D vertices.
+
+u = np.insert(verts3D, 3, 1, axis=1)
+v = np.insert(verts3D, 3, -1, axis=1)
+verts4D = np.append(u, v, axis=0)
+
+###############################################################################
+# We define a `rotate4D` function that takes 4D matrix as parameter and rotates
+# it in XY plane (Z axis) and ZW plane (an imaginary axis), projects it to the
+# 3D plane so that we can render it in a scene.
+
+
+def rotate4D(verts4D):
+    cos = np.cos(angle)
+    sin = np.sin(angle)
+    rotation4d_xy = [[cos, -sin, 0, 0],
+                     [sin, cos, 0, 0],
+                     [0, 0, 1, 0],
+                     [0, 0, 0, 1]]
+    rotation4d_zw = [[1, 0, 0, 0],
+                     [0, 1, 0, 0],
+                     [0, 0, cos, -sin],
+                     [0, 0, sin, cos]]
+    distance = 2
+    projected_marix = []
+    for vert in verts4D:
+        rotated_3D = np.dot(rotation4d_xy, vert)
+        rotated_3D = np.dot(rotation4d_zw, rotated_3D)
+        w = 1/(distance - rotated_3D[3])
+        proj_mat4D = np.array(
+            [[w, 0, 0, 0],
+             [0, w, 0, 0],
+             [0, 0, w, 0]]
+        )
+
+        projeced_mat3D = np.dot(proj_mat4D, rotated_3D)
+        projected_marix.append(projeced_mat3D)  # vertices to be proj (3D)
+
+    return np.array(projected_marix)
+
+###############################################################################
+# Now, We have 4D points projected to 3D. Let's define a function to connect
+# lines.
+
+
+def connect_points(verts3D):
+    lines = []
+    len_vert = len(verts3D)
+
+    for i in range(len_vert-1):
+        if i < 8:
+            lines.append(np.array([verts3D[i], verts3D[i+8]]))
+        if i == 7:
+            pass
+        else:
+            lines.append(np.array([verts3D[i], verts3D[i+1]]))
+        if i % 4 == 0:
+            lines.append(np.array([verts3D[i], verts3D[i+3]]))
+
+    for i in range(3):
+        lines.append(np.array([verts3D[i], verts3D[i+5]]))
+        lines.append(np.array([verts3D[i+8], verts3D[i+5+8]]))
```

Comment:
same comment as above.

We recommend avoiding using a list of `np.array` because we observe memory issues.

Can you update your code? You can create another `np.array` with an additional dimension.